src/service.rs

```diff
@@ -516,7 +522,7 @@ where
         &mut self,
         mut handle: H,
         remote_pubkey: Option<PublicKey>,
-        address: Multiaddr,
+        mut address: Multiaddr,
```

Comment:
Can we remove it?

Reply:
Because there is no peer id(p2p protocol) in the `multiaddr` that the listener gets, so here need to insert it here.